libs/core/langchain_core/utils/function_calling.py

```diff
@@ -311,3 +318,104 @@ def convert_to_openai_tool(
         return tool
     function = convert_to_openai_function(tool)
     return {"type": "function", "function": function}
+
+
+class Example(TypedDict):
+    """A representation of an example consisting of text input and expected tool calls.
+
+    For extraction, the tool calls are represented as instances of pydantic model.
+    """
+
+    input: str  # This is the example text
+    tool_calls: List[BaseModel]  # Instances of pydantic model that should be extracted
+
+
+def tool_example_to_messages(example: Example) -> List[BaseMessage]:
```

Comment:
does input need to be dict, why not just have `input` and `tool_calls` as args

Reply:
idk @eyurtsev any thoughts?

Reply:
yeah this can be unpacked